### PR TITLE
feat: blogs | blog filter option is added

### LINKED ISSUE #600: [FEATURE] Blogs | Add Filters Of Users
### Detailed description

https://dev.thecyberhub.org/#/blogs

Main page --> Blogs Tab

In the Blogs page add a filter so the user could see all the blogs or only "following users" blogs

### Context

Users could decide to see only blogs of users that they follow after.

### Possible implementation

Think about good place to put the filters. If you have a question, we can discuss.
image

### Additional information

The "following users" state can be found in src/features/follow , we use Redux.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by sumanjeet0012**
## Fixes Issue

Closes #600 

## Changes proposed

A new select option is created for selecting "all" blogs or "following" blogs.
The select option is added in the right side as shown in screenshot.

## Screenshots

image

## Code of Conduct

- [x] By submitting this pull request, I confirm I've read and complied with the [CoC](https://github.com/thecyberworld/Support/blob/main/CODE_OF_CONDUCT.md) 🖖

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

---

You can also join our [Discord](https://discord.gg/QHBPq6xP5p) community.
Feel free to check out other cool repositories of the [Thecyberworld](https://github.com/thecyberworld).
Join the Thecyberworld GitHub Organisation by raising an [issue](https://github.com/thecyberworld/Support/issues/new?assignees=&labels=invite+me+to+the+organisation&template=invitation.yml&title=Please+invite+me+to+the+GitHub+Community+Organization) (you will be sent an invitation).

### CODE REVIEW

**Comment by ArkadiK94** [comment]
@sumanjeet0012 could you add some margin because the filter is to much close to the blog posts?

**Comment by sumanjeet0012** [comment]
I have added margins to filter option.

File: src/components/Blogs/Blogs.jsx [hunk 1]
```diff
@@ -76,12 +105,14 @@ const Blogs = () => {
             <BlogsContainer>
                 <BlogsSection>
                     <RouterNavCreateButtonLink to={"/dashboard/blogs/create"}>Create Blog</RouterNavCreateButtonLink>
+                    <BlogFilter selectedBlogs={selectedBlogs} handleSelectedBlogs={handleSelectedBlogs} />
```
Comment (ArkadiK94) [inline src/components/Blogs/Blogs.jsx:108]: Please, render this filter only if the user is registered, in my opinion, unregistered users don't follow others so this useless to show this filter to them.

File: src/components/Blogs/Blogs.jsx [hunk 2]
```diff
@@ -21,22 +31,40 @@ const Blogs = () => {
         (state) => state.userDetail,
     );
 
+    const followUserId = user?._id;
```
Comment (ArkadiK94) [inline src/components/Blogs/Blogs.jsx:34]: This name is misleading, this is not a user id of who we follow . It is our user id. Which used to get info about users that we follow after.

File: src/components/Blogs/Blogs.jsx [hunk 3]
```diff
@@ -21,22 +31,40 @@ const Blogs = () => {
...
+            avatar,
+            verified,
+        };
+    });
 
+    const followingBlogData = blogs?.map((blog) => {
+        if (!followData?.following?.includes(blog.user)) {
+            return null;
+        }
+        const userDetail = userDetails?.find((user) => user.user === blog.user);
```
Comment (ArkadiK94) [inline src/components/Blogs/Blogs.jsx:67]:
```
const userDetail = userDetails?.find((user) => user.user === blog.user);
const { username, avatar, verified } = userDetail || {};

return {
    ...blog,
    username,
    avatar,
    verified,
};
```
You have this all code duplicated, so try to refactor the blogsData and followingBlogData by using external function of the duplicated of code.

**Review by ArkadiK94 — APPROVED** [review]
@sumanjeet0012 Great job, I have a few comments.

**Comment by ArkadiK94** [comment]
@sumanjeet0012 Could you make the adjustments from my comments?

File: src/components/Blogs/Blogs.jsx (hunk 1 above)
Comment (sumanjeet0012) [reply src/components/Blogs/Blogs.jsx]: It's done

File: src/components/Blogs/Blogs.jsx (hunk 2 above)
Comment (sumanjeet0012) [reply src/components/Blogs/Blogs.jsx]: followUserId is changed to userId

File: src/components/Blogs/Blogs.jsx (hunk 3 above)
Comment (sumanjeet0012) [reply src/components/Blogs/Blogs.jsx]: userDetailsFunction is created and duplicate codes are removed.

**Comment by sumanjeet0012** [comment]
I haven't seen your comments, today I noticed it when I received mail from GitHub.

ping me on discord, if you have any suggestions, I will try to do it on the same day.

**Comment by ArkadiK94** [comment]
@sumanjeet0012 Thanks, looks great 👍

**Pull request merged**